src/components/ChatInput/ChatInput.tsx

```diff
@@ -120,30 +122,79 @@ export interface ChatInputProps {
   onSubmit: (message: string) => void;
   accessibility?: TockAccessibility;
   clearMessages: () => void;
+  autoCompletionEndPoint?: string;
+  minValueLength: number
 }
 
 const ChatInput: (props: ChatInputProps) => JSX.Element = ({
   disabled,
   onSubmit,
   accessibility,
   clearMessages,
+  autoCompletionEndPoint,
+  minValueLength
 }: ChatInputProps): JSX.Element => {
   const [value, setValue] = useState('');
-  const submit = (event: FormEvent<HTMLFormElement>) => {
+  const [suggestions, setSuggestions] = useState<string[]>([]);
+  const [suggestionSelected, setSuggestionSelected] = useState(false);
+
+    const submit = (event: FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     if (value) {
       onSubmit(value);
       setValue('');
+      setSuggestionSelected(false);
     }
   };
 
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                if (value.length >= minValueLength && !suggestionSelected) {
+                    // Fetch data from the autoCompletionEndPoint/autocompletion.json - endpoint
+                    //autoCompletionEndPoint : 'https://example.fr'
+                    const response = await fetch(`${autoCompletionEndPoint}/autocompletion.json`); //
+                    const data = await response.json();
+                    const filteredSuggestions = data.filter((item: string) =>
+                        item.toLowerCase().includes(value.toLowerCase())
+                    );
+                    setSuggestions(filteredSuggestions);
+                } else {
+                    setSuggestions([]);
+                }
+            } catch (error) {
+                console.error('Error fetching suggestions:', error);
+            }
+        };
+        fetchData();
+    }, [value, minValueLength, suggestionSelected]);
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setValue(e.target.value);
+        setSuggestionSelected(false);
+    };
+
+    const handleSuggestionClick = (suggestion: string) => {
+        setValue(suggestion);
+        setSuggestions([]);
+        setSuggestionSelected(true);
+    };
```

Comment:
Why go through a `useEffect` and force intermediary re-renders, as well as require additional state management, rather than directly perform the query asynchronously in the `onChange` handler?